### PR TITLE
fix: Explain error thrown when cypress commands in .should() callback

### PR DESCRIPTION
**Pull request by BlueWinds**
Associated code: https://github.com/cypress-io/cypress/pull/23734

Beginning in Cypress 11, invoking Cypress commands inside `.should()` will throw an error. Doing this currently (Cy 10 and earlier) results in undefined behavior - a term which should scare anyone familiar with it from other programming languages.

We're resolving this by throwing an error, disallowing the situation entirely. This is a breaking change, which is why it's going out in 11.0.0.

### CODE REVIEW

**Review by lmiller1990 — APPROVED** [review]
Nice. Let's hold off merging until 11.0 is out.

File: content/api/commands/should.md [hunk 1]
```diff
@@ -172,6 +172,31 @@ Be sure _not_ to include any code that has side effects in your callback
 function. The callback function will be retried over and over again until no
 assertions within it throw.
 
+You cannot invoke Cypress commands inside of a `.should()` callback function.
+Use cypress commands before or after `.should()` instead.
```
Comment (rachelruderman) [inline content/api/commands/should.md:176]:
```suggestion
Use Cypress commands before or after `.should()` instead.
```

**Review by rachelruderman — APPROVED** [review]

**Comment by flotwig** [comment]
@lmiller1990 @BlueWinds I'd just go ahead and get an 11.0.0 branch/PR up, so we don't have to keep track of this until then. It might be a while til it's ready since I believe we're waiting on multidomain GA to drop 11.0.0.

File: content/api/commands/should.md [hunk 2]
```diff
@@ -172,6 +172,31 @@ Be sure _not_ to include any code that has side effects in your callback
...
 
+You cannot invoke Cypress commands inside of a `.should()` callback function.
+Use Cypress commands before or after `.should()` instead.
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```javascript
+cy.get('p').should(($p) => {
+  cy.log($p)
+  ...
```
Comment (flotwig) [inline content/api/commands/should.md:183]:
```suggestion
  // ...
```

**Review by flotwig — APPROVED** [review]

File: content/api/commands/should.md [hunk 3]
```diff
@@ -172,6 +172,31 @@ Be sure _not_ to include any code that has side effects in your callback
...
+})
+```
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```javascript
+cy.get('p').should(($p) => { ... }).log()
```
Comment (BlueWinds) [inline content/api/commands/should.md:190]:
```suggestion
cy.get('p').should(($p) => {
  // ...
}).log()
```

File: content/api/commands/should.md [hunk 4]
```diff
@@ -172,6 +172,31 @@ Be sure _not_ to include any code that has side effects in your callback
...
+
+// or
+
+cy.get('p').then(($p) => {
+  ...
+  cy.log($p)
+  })
```
Comment (BlueWinds) [inline content/api/commands/should.md:197]:
```suggestion
  // ...
  cy.log($p)
})
```

**Pull request merged**